example_client/cpp/src/resnet_client_benchmark.cpp

```diff
@@ -0,0 +1,452 @@
+        auto it = outputs.find(this->config.outputName);
+        if (it == outputs.end()) {
+            std::cout << "cannot find output " << this->config.outputName << std::endl;
+            return false;
+        }
+        tensorflow::TensorProto& resultTensorProto = it->second;
+        tensorflow::Tensor tensor;
+        bool converted = tensor.FromProto(resultTensorProto);
+        if (!converted) {
+            std::cout << "the result tensor[" << it->first << "] convert failed." << std::endl;
+            return false;
+        }
+        predictedLabels = argmax(tensor);
+        return true;
+    }
+
+    bool schedulePredict(tensorflow::int64 iteration) {
+        PredictResponse response;
+        ClientContext context;
+
+        auto* call = new AsyncClientCall<T>;
+        call->id = iteration + 1;
+
+        if (this->config.benchmark_mode == 0) {
+            // Pre-processing step.
+            // Packing image into gRPC message.
+            PredictRequest request;
+            request.mutable_model_spec()->set_name(this->config.modelName);
+            request.mutable_model_spec()->set_signature_name("serving_default");
+            google::protobuf::Map<tensorflow::string, tensorflow::TensorProto>& inputs = *request.mutable_inputs();
+            call->selectedEntries = selectEntries(this->entries, this->config.batchSize, iteration);
+            this->prepareBatchedInputs(inputs, call->selectedEntries);
+            call->response_reader = stub_->PrepareAsyncPredict(&call->context, request, &this->cq_);
+            call->response_reader->StartCall();
+            call->response_reader->Finish(&call->reply, &call->status, (void*)call);
+        } else {
+            // No pre-processing step.
+            // Re-use previously prepared gRPC message.
+            call->response_reader = stub_->PrepareAsyncPredict(&call->context, this->predictRequest, &this->cq_);
+            call->response_reader->StartCall();
+            call->response_reader->Finish(&call->reply, &call->status, (void*)call);
+        }
+
+        if (config.benchmark_mode == 0) {
+            std::cout << "Scheduled request no. " << call->id << std::endl;
+        }
+        return true;
+    }
+
+    void asyncCompleteRpc() {
+        void* got_tag;
+        bool ok = false;
+
+        while (cq_.Next(&got_tag, &ok)) {
+            if (++this->finishedIterations >= this->config.iterations * this->config.producers) {
+                cq_.Shutdown();
+            }
+            cv.notify_one();
+            auto* call = static_cast<AsyncClientCall<T>*>(got_tag);
+            ResourceGuard guard(call);
+
+            if (config.benchmark_mode == 0) {
+                std::cout << "Received response no. " << call->id << std::endl;
+            }
+            auto& response = call->reply;
+
+            if (!ok) {
+                std::cerr << "Request is not ok" << std::endl;
+                continue;
```

Comment:
I wonder if we should continue with the next requests in case of error. For benchmark mode we won't get real inference results if many of them fail before inference happens. For non-benchmark we don't get full and correct information about accuracy. On the other hand users can see those logs, so they would be aware that something is wrong.

Reply:
added warning when at least one inference fails